### ISSUE TITLE
Same system UUID shared between nodes

### ISSUE THREAD

**Issue by fabiand** [kubernetes-sigs/kind#1027]
**What happened**:
I tried to run a VM live migration on a kind cluster using kubevirt VMs.
The error I got was:
```
 Warning  Migrated          4s                virt-handler, kind-worker   VirtualMachineInstance migration uid 5a74fa0a-d2ec-48b4-a39e-e2659ae10602 failed. reason:Live migration failed virError(Code=1, Domain=10, Message='internal error: Attempt to migrate guest to the same host 1ad82a4c-2a18-11b2-a85c-82c3099c6263')
```

Investigating this showed that indeed all nodes share the same UUID, this should not be the case.
KubeVirt is one component noticing this, but there might be more .

**What you expected to happen**:
Migration completes because all nodes have different UUIDs.

**How to reproduce it (as minimally and precisely as possible)**:
Can be provided if necessary.

**Anything else we need to know?**:
KubeVirt is correct:
```
$ k get nodes -o yaml | grep UUID
      systemUUID: 1ad82a4c-2a18-11b2-a85c-82c3099c6263
      systemUUID: 1ad82a4c-2a18-11b2-a85c-82c3099c6263
      systemUUID: 1ad82a4c-2a18-11b2-a85c-82c3099c6263
```

**Environment:**

- kind version: (use `kind version`): 0.5.1
- Kubernetes version: (use `kubectl version`):
- Docker version: (use `docker info`):
- OS (e.g. from `/etc/os-release`):

**Comment by fabiand** [comment]
Ok, this value comes from the physical host:

```
$dmidecode | grep d82a
	UUID: 1ad82a4c-2a18-11b2-a85c-82c3099c6263
```

**Comment by fabiand** [comment]
Can also be retrieved by `dmidecode --string system-uuid`

**Comment by fabiand** [comment]
Which is also accesible here:
- /sys/devices/virtual/dmi/id/product_uuid
- /sys/class/dmi/id/product_uuid

According to k8s `GetSystemUUID` it might a solution to actual put a random uuid into a file and then mounting them over the product_uuid paths above.

**Comment by BenTheElder** [comment]
Hmm so we do ensure a unique ID under systems and we fake out the product name already so there is precedent. https://github.com/kubernetes-sigs/kind/blob/7a99ef875be750625b1335f1b9ab159762c27841/images/base/files/usr/local/bin/entrypoint#L78

**Comment by BenTheElder** [comment]
if we can figure out a good way to generate a valid value we can do a bind
mount the same way we handle the product name

**Comment by BenTheElder** [comment]
for future reference:
https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/install-kubeadm/#verify-the-mac-address-and-product-uuid-are-unique-for-every-node

**Comment by aojea** [comment]
/close 
fix was merged in #1028
